graph/src/blockchain/mod.rs

```diff
@@ -189,7 +191,7 @@ pub trait Blockchain: Debug + Sized + Send + Sync + Unpin + 'static {
         deployment: DeploymentLocator,
         store: impl DeploymentCursorTracker,
         start_blocks: Vec<BlockNumber>,
-        source_subgraph_stores: Vec<(DeploymentHash, Arc<dyn ReadStore>)>,
+        source_subgraph_stores: Vec<(DeploymentHash, Arc<dyn SourceableStore>)>,
```

Comment:
You don't necessarily need the `DeploymentHash` here since `SourceableStore.input_schema().id()` will return it (and there could be a convenience method on `SourceableStore` for it)

Actually, not having it explicit lowers the risk that there is a mismatch between the two.